src/ipld/json.rs

```diff
@@ -240,51 +240,23 @@ impl<'de> de::Visitor<'de> for JSONVisitor {
 
 #[cfg(test)]
 mod tests {
-    use cid::Cid;
     use quickcheck_macros::quickcheck;
     use serde_json;
 
     use super::*;
 
-    #[derive(Clone, Debug, PartialEq)]
-    struct IpldWrapper {
-        ipld: libipld_macro::Ipld,
-    }
-
-    impl quickcheck::Arbitrary for IpldWrapper {
-        fn arbitrary(g: &mut quickcheck::Gen) -> Self {
-            let bool_item = bool::arbitrary(g);
-            let int_item = i128::arbitrary(g);
-            let float_item = f64::arbitrary(g);
-            let string_item = String::arbitrary(g);
-            let bytes_item = Vec::arbitrary(g);
-            let list_item = Vec::new();
-            let map_item = BTreeMap::new();
-            let cid = Cid::new_v1(
-                u64::arbitrary(g),
-                cid::multihash::Multihash::wrap(u64::arbitrary(g), &[u8::arbitrary(g)]).unwrap(),
-            );
-            let binding = [
-                Ipld::Null,
-                Ipld::Bool(bool_item),
-                Ipld::Integer(int_item),
-                // Filter out problematic NaN values.
-                Ipld::Float(if float_item.is_nan() { 0.0 } else { float_item }),
-                Ipld::String(string_item),
-                Ipld::Bytes(bytes_item),
-                Ipld::List(list_item),
-                Ipld::Map(map_item),
-                Ipld::Link(cid),
-            ];
-            let ipld = g.choose(&binding).unwrap();
-            IpldWrapper { ipld: ipld.clone() }
-        }
-    }
-
     #[quickcheck]
-    fn ipld_roundtrip(ipld: IpldWrapper) {
-        let serialized: String = serde_json::to_string(&IpldJsonRef(&ipld.ipld)).unwrap();
+    fn ipld_roundtrip(ipld: Ipld) {
+        // Filter out problematic NaN values.
```

Comment:
I'd say something like `Reject Ipld input containing NaN floats`. In my mind, filtering out values would mean returning a new `Ipld` structure without the problematic floats.

Reply:
Also, write a note about _why_ you're rejecting NaN input.